spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/sql/TestFilterPushDown.java

```diff
@@ -578,6 +578,28 @@ public void testFilterPushdownWithSpecialFloatingPointPartitionValues() {
         ImmutableList.of(row(4, Double.NEGATIVE_INFINITY)));
   }
 
+  @TestTemplate
+  public void testFilterPushdownWithRequiredNestedFieldInOptionalStruct() {
+    sql(
+        "CREATE TABLE %s (id INT NOT NULL, address STRUCT<street: STRING NOT NULL>)"
+            + "USING iceberg ",
+        tableName);
+    configurePlanningMode(planningMode);
+
+    sql("INSERT INTO %s VALUES (0, NULL)", tableName);
+    sql("INSERT INTO %s VALUES (1, STRUCT('123 Main St'))", tableName);
+
+    checkOnlyIcebergFilters(
+        "address is null" /* query predicate */,
```

Comment:
we should also test predicate on nested field, right? that is what this PR tries to fix.
```
address.street is null
```